src/array_api_stubs/_draft/array_object.py

```diff
@@ -620,9 +622,13 @@ def __getitem__(
         ----------
         self: array
             array instance.
-        key: Union[int, slice, ellipsis, None, Tuple[Union[int, slice, ellipsis, None], ...], array]
+        key: Union[int, slice, ellipsis, None, SupportsIndex, Tuple[Union[int, slice, ellipsis, None, SupportsIndex], ...], array]
             index key.
 
+
+        .. note::
+           ``key`` can only be an array if it is valid for boolean array indexing, or supports ``__index__()``.
```

Comment:
Presumably somewhere else we say it should support `__index__` if and only if it is a 0-D integer array. Maybe it would be clearer to just say that directly here.

Reply:
Yep that'd help, let me know on the wording from my latest commit.